cognite/neat/rules/models/_rules/dms_architect_rules.py

```diff
@@ -701,29 +709,39 @@ def _create_views_with_node_types(
                 prop_name = to_camel(prop.view_property) if self.standardize_casing else prop.view_property
                 view.properties[prop_name] = view_property
 
+        dms_view_by_view_id = {
+            view.view.as_id(default_space, default_version, self.standardize_casing): view for view in dms_views
+        }
         node_types = dm.NodeApplyList([])
-        parent_views = {parent for view in views for parent in view.implements or []}
-        node_type_flag = False
+        {parent for view in views for parent in view.implements or []}
+
         for view in views:
-            ref_containers = view.referenced_containers()
-            has_data = dm.filters.HasData(containers=list(ref_containers)) if ref_containers else None
-            node_type = dm.filters.Equals(["node", "type"], {"space": view.space, "externalId": view.external_id})
-            if view.as_id() in parent_views:
-                view.filter = has_data
-            elif has_data is None:
-                # Child filter without container properties
-                if node_type_flag:
-                    # Transformations do not yet support setting node type.
-                    view.filter = node_type
-                    node_types.append(dm.NodeApply(space=view.space, external_id=view.external_id, sources=[]))
-            else:
-                # Child filter with its own container properties
-                if node_type_flag:
-                    # Transformations do not yet support setting node type.
-                    view.filter = dm.filters.And(has_data, node_type)
-                    node_types.append(dm.NodeApply(space=view.space, external_id=view.external_id, sources=[]))
+            selected_filters = dms_view_by_view_id[view.as_id()].filter_
+
+            filters: list[dm.filters.Filter] = []
+            for selected_filter in selected_filters or ["hasData"]:
```

Comment:
Probably misreading the code but this at first reading looks like you can add either "hasData" or "nodeType" but not both.

Reply:
It is clearly a list, which is appended to, so not really sure how to make it much clearer that this is a set of filters.

Reply:
Did a small trick now that hopefully makes this clearer.